### ISSUE TITLE
[🐛 BUG]: Local activities run on workflow worker pool

### ISSUE THREAD

**Issue by Zylius** [roadrunner-server/roadrunner#1940]
### No duplicates 🥲.

- [X] I have searched for a similar issue in our bug tracker and didn't find any solutions.

### What happened?

RoadRunner runs all local activities on workflow worker pool, instead of activity pool. This causes the configuration for temporal.activities.allocate_timeout to be ignored, and some local activities do eventually fail/timeout waiting for a free worker.

### Version (rr --version)

2023.3.12

### How to reproduce the issue?

Just launch a workflow with a local activity and watch only the workflow worker execution count rise.

### Relevant log output

```shell
Workers of [temporal]:
+---------+-----------+---------+---------+---------+--------------------+
|   PID   |  STATUS   |  EXECS  | MEMORY  |  CPU%   |      CREATED       |
+---------+-----------+---------+---------+---------+--------------------+
|      43 | ready     |       0 | 122 MB  |    0.07 | 32 minutes ago     |
|      70 | ready     | 129,747 | 280 MB  |   38.44 | 32 minutes ago     |
+---------+-----------+---------+---------+---------+--------------------+
~/src $ 
```
```

**Comment by rustatian** [comment]
Hey @Zylius 👋 
Thanks for the report. Will be fixed this Thursday in `v2024.1.4`.

**Comment by rustatian** [comment]
Fixed
